src/ucs/memory/rcache.c

```diff
@@ -185,23 +193,77 @@ static ucs_mpool_ops_t ucs_rcache_mp_ops = {
     .obj_cleanup   = NULL
 };
 
+static unsigned ucs_rcache_region_page_count(ucs_rcache_region_t *region)
+{
+    return (region->super.end - region->super.start) / ucs_get_page_size();
+}
+
+static ucs_status_t ucs_rcache_validate_pfn(ucs_rcache_t *rcache,
+                                            ucs_rcache_region_t *region,
+                                            unsigned page_num,
+                                            unsigned long region_pfn,
+                                            unsigned long actual_pfn)
+{
+    if (region_pfn != actual_pfn) {
+        ucs_rcache_region_error(rcache, region, "pfn check failed");
+        ucs_fatal("%s: page at virtual address 0x%lx moved from pfn 0x%lx to pfn 0x%lx",
+                  rcache->name,
+                  region->super.start + (page_num * ucs_get_page_size()),
+                  region_pfn, actual_pfn);
+        return UCS_ERR_NO_MESSAGE;
```

Comment:
what is the use of return value, since if the check fail is fatal it does not return